### PR TITLE
[FEA] Synthetic dataloader for weather prediction models

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/datapipes/climate/dummy.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            dataset=dataset,
+            batch_size=kwargs.get("batch_size", 1),
+            shuffle=kwargs.get("shuffle", False),
+            num_workers=kwargs.get("num_workers", 0),
+            pin_memory=kwargs.get("pin_memory", False),
+            drop_last=kwargs.get("drop_last", False),
+        )
+
+
+class DummyWeatherDataset(Dataset):
```
Comment (akshaysubr) [inline modulus/datapipes/climate/dummy.py:44]: Would prefer renaming to `SyntheticWeatherDataset` maybe

File: modulus/datapipes/climate/dummy.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                )
+        return daily_temps
+
+    def __len__(self) -> int:
+        """
+        Returns the number of samples available in the dataset.
+        """
+        return self.num_days - self.num_steps
+
+    def __getitem__(self, idx: int) -> torch.Tensor:
```
Comment (akshaysubr) [inline modulus/datapipes/climate/dummy.py:141]: This needs a test since it is a utility in the main repo

**Review by akshaysubr — CHANGES_REQUESTED** [review]

File: modulus/datapipes/climate/dummy.py (hunk 1 above)
Comment (mnabian) [reply modulus/datapipes/climate/dummy.py]: Done

File: modulus/datapipes/climate/dummy.py (hunk 2 above)
Comment (mnabian) [reply modulus/datapipes/climate/dummy.py]: Added tests

**Review by stadlmax — APPROVED** [review]
a minor nitpick, otherwise looks good to me

File: modulus/datapipes/climate/dummy.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        noise_level (float): Standard deviation of the noise added to temperature data.
+        **kwargs: Additional keyword arguments for advanced configurations.
+    """
+
+    def __init__(
+        self,
+        channels: List[int],
+        num_samples_per_year: int,
+        num_steps: int,
+        device: str = "cuda",
```
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/datapipes/climate/dummy.py]: Fixed it, thanks

File: modulus/datapipes/climate/synthetic.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        im = ax.imshow(temp_data, cmap="coolwarm", interpolation="nearest")
+        ax.set_title(f"Channel {i+1}")
+        ax.axis("off")
+
+    fig.colorbar(im, ax=axes.ravel().tolist(), shrink=0.6)
+    plt.suptitle("Temperature Distribution Across Different Atmospheric Layers")
+    plt.savefig("temperature_data.png")
+
+
+if __name__ == "__main__":
```
Comment (loliverhennigh) [inline modulus/datapipes/climate/synthetic.py:178]: Maybe remove this part in favor of the unit test

**Review by loliverhennigh — APPROVED** [review]
Looks good to me but added one comment.

File: modulus/datapipes/climate/synthetic.py (hunk 4 above)
Comment (mnabian) [reply modulus/datapipes/climate/synthetic.py]: Done

**Review by akshaysubr — APPROVED** [review]

**Pull request merged**